### PR TITLE
chore: fix more tests (state manager, consensus), fix serialization

### PR DESCRIPTION
**Pull request by lmoe**

### CODE REVIEW

File: clients/iota-go/iotago/iotatest/random.go [hunk 1]
```diff
@@ -36,8 +36,8 @@ func RandomTransactionData() *iotago.TransactionData {
 				Package:       RandomAddress(),
 				Module:        "test_module",
 				Function:      "test_func",
-				TypeArguments: []iotago.TypeTag{},
-				Arguments:     []iotago.Argument{},
+				TypeArguments: nil,
```
Comment (howjmay) [inline clients/iota-go/iotago/iotatest/random.go:39]: These fields can be neglected

File: packages/chain/statemanager/sm_gpa/sm_gpa_utils/test_block_factory.go [hunk 2]
```diff
@@ -59,14 +59,14 @@ func NewBlockFactory(t require.TestingT, chainInitParamsOpt ...BlockFactoryCallA
 		originBlock.L1Commitment(),
 		&iotago.ObjectID{},
 		gas.DefaultFeePolicy(),
-		isc.NewCallArguments(),
+		chainInitParams,
 		"",
 	)
 	originAnchorData := iscmove.RefWithObject[iscmove.Anchor]{
 		ObjectRef: iotago.ObjectRef{
```
Comment (howjmay) [inline packages/chain/statemanager/sm_gpa/sm_gpa_utils/test_block_factory.go:66]: Maybe use RandomObjectRef?

**Review by howjmay — APPROVED** [review]

File: clients/iota-go/iotago/iotatest/random.go (hunk 1 above)
Comment (lmoe) [reply clients/iota-go/iotago/iotatest/random.go:39]: Nah it's required here as the BCS testing funcs differ between a nil and an empty array :D

File: packages/chain/statemanager/sm_gpa/sm_gpa_utils/test_block_factory.go (hunk 2 above)
Comment (lmoe) [reply packages/chain/statemanager/sm_gpa/sm_gpa_utils/test_block_factory.go:66]: We need to set the proper ObjectID == ChainID here :)

**Pull request merged**